### PR TITLE
[Feature]Support refresh mv by partition

### PR DESCRIPTION
**Pull request by QingdongZeng3**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#11445

```
CREATE TABLE t1 (
                  col1 int,
                  col2 date,
                  col3 varchar(20)
                ) ENGINE=OLAP
                DUPLICATE KEY(col1,col2)
                COMMENT "OLAP"
                PARTITION BY RANGE(`col2`)
                (
                       START ("2021-01-01") END ("2021-01-04") EVERY (INTERVAL 1 day) 
                )
                DISTRIBUTED BY HASH(col1) BUCKETS 3
                PROPERTIES (
                "replication_num" = "3"
                );


CREATE MATERIALIZED VIEW mv1 PARTITION BY `col2`
DISTRIBUTED BY HASH(col1) BUCKETS 3
REFRESH manual
AS SELECT col1, col2
FROM t1;

refresh materialized view mv1 force;

insert into t1  values (1,'2021-01-01','sss'),(2,'2021-01-01','sss'),(3,'2021-01-02','sss');

MySQL [p1]> select * from t1;
+------+------------+------+
| col1 | col2       | col3 |
+------+------------+------+
|    2 | 2021-01-01 | sss  |
|    3 | 2021-01-02 | sss  |
|    1 | 2021-01-01 | sss  |
+------+------------+------+
3 rows in set (0.01 sec)

refresh materialized view mv1 partition START ("2021-01-01") END ("2021-01-02") force;

select * from mv1;
+------+------------+
| col1 | col2       |
+------+------------+
|    2 | 2021-01-01 |
|    1 | 2021-01-01 |
+------+------------+
2 rows in set (0.01 sec)

refresh materialized view mv1 ;

MySQL [p1]> select * from mv1;
+------+------------+
| col1 | col2       |
+------+------------+
|    3 | 2021-01-02 |
|    1 | 2021-01-01 |
|    2 | 2021-01-01 |
+------+------------+
3 rows in set (0.01 sec)


```

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by Astralidea** [comment]
run all

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 78 / 101 (77.23%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[179]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|22|36.36%|[3444, 3452, 3453, 3454, 3455, 3456, 3457, 3458, 3460, 3461, 3462, 3463, 3464, 3465]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|14|16|87.50%|[376, 377]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionWithoutIntervalDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 1]
```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
...
+    private Set<String> getPartitionsToRefreshForMaterializedView(Map<String, String> properties)
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
+        Set<String> rangePartitionNames = null;
+        if (StringUtils.isEmpty(start) && StringUtils.isEmpty(end) && force) {
+            LOG.debug("materialized view force to refresh all partitions.");
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:376]: nit: materialized-> Materialized

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 2]
```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
...
+            return materializedView.getPartitionNames();
+        }
+        if (!StringUtils.isEmpty(start) && !StringUtils.isEmpty(end)) {
+            rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
+            LOG.debug("materialized view refresh partitions {}", rangePartitionNames);
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:381]: ditto

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 3]
```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
         return false;
     }
 
-    private Set<String> getPartitionsToRefreshForMaterializedView() {
+    private Set<String> getPartitionsToRefreshForMaterializedView(Map<String, String> properties)
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:371]: should we validate partition info for `start` and `end`?

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 4]
```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
...
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
+        Set<String> rangePartitionNames = null;
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:374]: if we only specify start or end then we should refresh all the partitions right?

Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:374]: rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
and check start and end in getPartitionNamesByRange

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshMaterializedViewStatement.java [hunk 5]
```diff
@@ -5,9 +5,15 @@
 
 public class RefreshMaterializedViewStatement extends DdlStmt {
     private final TableName mvName;
+    private final RangePartitionWithoutIntervalDesc rangePartitionWithoutIntervalDesc;
+    private final boolean force;
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshMaterializedViewStatement.java:9]: should use forceRefresh

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 3 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:371]: yes,  see 
 com.starrocks.sql.analyzer.MaterializedViewAnalyzer.MaterializedViewAnalyzerVisitor#visitRefreshMaterializedViewStatement

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 4 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:374]: if we only specify start or end then throw exception when sql analyze.

Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:374]: check  in 
com.starrocks.sql.analyzer.MaterializedViewAnalyzer.MaterializedViewAnalyzerVisitor#visitRefreshMaterializedViewStatement

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 2 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshMaterializedViewStatement.java (hunk 5 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/RefreshMaterializedViewStatement.java]: ok

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 1 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: done

**Comment by QingdongZeng3** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 99 / 135 (73.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[179]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|22|36.36%|[3444, 3452, 3453, 3454, 3455, 3456, 3457, 3458, 3460, 3461, 3462, 3463, 3464, 3465]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|21|34|61.76%|[544, 545, 547, 554, 555, 559, 560, 561, 562, 564, 565, 566, 568]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|14|16|87.50%|[376, 377]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionWithoutIntervalDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|20|20|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 6]
```diff
@@ -364,14 +365,30 @@ private boolean needToRefreshNonPartitionTable() {
...
+        if (StringUtils.isEmpty(start) && StringUtils.isEmpty(end) && force) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:375]: remove this

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 4 above)
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:374]:
```suggestion
        Set<String> rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
```

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 6 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: ok

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by QingdongZeng3** [comment]
run starrocks_admit_test

**Comment by QingdongZeng3** [comment]
run starrocks_fe_unittest

**Comment by QingdongZeng3** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 98 / 132 (74.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[179]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|22|36.36%|[3444, 3452, 3453, 3454, 3455, 3456, 3457, 3458, 3460, 3461, 3462, 3463, 3464, 3465]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|21|34|61.76%|[544, 545, 547, 554, 555, 559, 560, 561, 562, 564, 565, 566, 568]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionWithoutIntervalDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 7]
```diff
@@ -1847,6 +1847,10 @@ multiRangePartition
     | START '(' string ')' END '(' string ')' EVERY '(' INTEGER_VALUE ')'
     ;
 
+rangePartitionWithoutIntervalDesc
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1850]: partitionRange is better?

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java [hunk 8]
```diff
@@ -172,6 +173,11 @@ default void refreshMaterializedView(String dbName, String mvName, int priority)
             throws DdlException, MetaNotFoundException {
     }
 
+    default void refreshMaterializedView(RefreshMaterializedViewStatement refreshMaterializedViewStatement,
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:176]: if you add this function, is the one void refreshMaterializedView(String dbName, String mvName, int priority) useful?

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 9]
```diff
@@ -404,6 +413,9 @@ private Set<String> getPartitionsToRefreshForMaterializedView() {
         } else {
             throw new DmlException("unsupported partition info type:" + partitionInfo.getClass().getName());
         }
+        if (null != rangePartitionNames) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:416]:
```suggestion
        if (rangePartitionNames != null) {
```

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 10]
```diff
@@ -3448,6 +3438,33 @@ public void cancelRefreshMaterializedView(String dbName, String mvName) throws D
...
+    @Override
+    public void refreshMaterializedView(RefreshMaterializedViewStatement refreshMaterializedViewStatement, int priority)
+            throws DdlException, MetaNotFoundException {
+        String dbName = refreshMaterializedViewStatement.getMvName().getDb();
+        String mvName = refreshMaterializedViewStatement.getMvName().getTbl();
+        boolean force = refreshMaterializedViewStatement.isForceRefresh();
+        RangePartitionWithoutIntervalDesc range =
+                refreshMaterializedViewStatement.getRangePartitionWithoutIntervalDesc();
+        if (range == null && !force) {
+            refreshMaterializedView(dbName, mvName, priority);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3453]: I think you can unify the logic of if and else into the same function. It is ambiguous here

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java (hunk 8 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMetadata.java:176]: RefreshMaterializedView may used by other inner function，it is not suitable to wrap parameters to statement in inner function，so I remain refreshMaterializedView(String dbName, String mvName, int priority) method ，to refresh all partitions and not force。

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 10 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java]: yes, It is ambiguous here

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 9 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: ok

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 98 / 130 (75.38%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[179]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|20|40.00%|[3444, 3452, 3453, 3454, 3455, 3456, 3457, 3458, 3459, 3460, 3461, 3462]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|21|34|61.76%|[544, 545, 547, 554, 555, 559, 560, 561, 562, 564, 565, 566, 568]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/PartitionRangeDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 4 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: ok

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 11]
```diff
@@ -364,14 +364,23 @@ private boolean needToRefreshNonPartitionTable() {
...
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
+        Set<String> rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
+        LOG.debug("Materialized view refresh partitions: {}", rangePartitionNames);
+        if (force) {
+            return rangePartitionNames;
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:376]: if force is true, it is also necessary to go through the mapping logic of calcPotentialRefreshPartition. maybe version map should not be taken into account.

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 11 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:376]: I agree

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 11 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java]: ok

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 116 / 150 (77.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[178]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|8|20|40.00%|[3444, 3452, 3453, 3454, 3455, 3456, 3457, 3458, 3459, 3460, 3461, 3462]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|21|34|61.76%|[544, 545, 547, 554, 555, 559, 560, 561, 562, 564, 565, 566, 568]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|29|31|93.55%|[373, 394]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/PartitionRangeDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java [hunk 12]
```diff
@@ -364,49 +364,81 @@ private boolean needToRefreshNonPartitionTable() {
...
-                    needRefreshMvPartitionNames.addAll(mvContext.baseToMvNameRef.get(basePartitionName));
-                }
+                return getMVPartitionNamesToRefreshByRangePartitionNamesAndForce(mvRangePartitionNames, force);
             } else if (partitionExpr instanceof FunctionCallExpr) {
-                // check if there is a load in the base table and add it to the refresh candidate
-                Set<String> baseChangedPartitionNames = materializedView.getNeedRefreshPartitionNames(partitionTable);
-                for (String baseChangedPartitionName : baseChangedPartitionNames) {
-                    needRefreshMvPartitionNames.addAll(mvContext.baseToMvNameRef.get(baseChangedPartitionName));
-                }
+                needRefreshMvPartitionNames = getMVPartitionNamesToRefreshByRangePartitionNamesAndForce(
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:396]: sorry. calcPotentialRefreshPartition has the logic of mapping. here just collecting the original mv partitions names is ok. The mapping logic here is not necessary.

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java (hunk 12 above)
Comment (QingdongZeng3) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java:396]: yes, there is no mapping logic, just collecting the original mv partitions names  by range and force.

**Comment by wanpengfei-git** [comment]
BE Build Success

**Comment by QingdongZeng3** [comment]
run starrocks_fe_unittest

**Comment by QingdongZeng3** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 114 / 148 (77.03%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ConnectorMetadata.java|0|1|00.00%|[178]|
|:large_blue_circle:|com/starrocks/qe/DDLStmtExecutor.java|0|1|00.00%|[261]|
|:large_blue_circle:|com/starrocks/scheduler/ExecuteOption.java|1|6|16.67%|[21, 22, 23, 24, 25]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|6|18|33.33%|[3462, 3463, 3464, 3465, 3466, 3467, 3468, 3469, 3470, 3471, 3472, 3474]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|21|34|61.76%|[544, 545, 547, 554, 555, 559, 560, 561, 562, 564, 565, 566, 568]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|29|31|93.55%|[386, 407]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskRunBuilder.java|15|15|100.00%|[]|
|:large_blue_circle:|com/starrocks/scheduler/TaskManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/PartitionRangeDesc.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RefreshMaterializedViewStatement.java|5|5|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**